controllers/machinehealthcheck_controller.go

```diff
@@ -403,19 +403,23 @@ func (r *MachineHealthCheckReconciler) remediate(target resources.Target, rm res
 	if err != nil {
 		return errors.Wrapf(err, "failed to get remediation template")
 	}
-	remediationCR, err := rm.GenerateRemediationCRForMachine(target.Machine, target.MHC, template)
-	if err != nil {
-		return errors.Wrapf(err, "failed to generate remediation CR")
-	}
 
 	// TODO add control plane label
 
 	// create remediation CR
-	var nodeName *string
+	var nodeNamePtr *string
```

Comment:
you can continue to use the pointer var only, and use `k8s.io/utils/pointer.StringDeref` for `GenerateRemediationCRForMachine()`, would be nicer than having 2 vars for the same thing IMHO.